.gitlab-ci.yml

```diff
@@ -25,12 +25,44 @@ stages:
   - deploy
   - notify
 
+########################################################################################################################
+# Configuration helpers
+########################################################################################################################
+
+.base-configuration:
+  tags: ['runner:main', 'size:large']
+  image: $CI_IMAGE
+
+.feature-branch-configuration:
+  extends: .base-configuration
+  except:
+    refs:
+      - main
+      - tags
+      - /^staging-[0-9]+$/
+      - /^release\/v[0-9.]+$/
+      - schedules
+
+.test-configuration:
+  stage: test
+  extends: .base-configuration
+  except:
+    refs:
+      - /^release\/v[0-9.]+$/
+      - schedules
+
+########################################################################################################################
+# CI image
+########################################################################################################################
+
 ci-image:
   stage: ci-image
   when: manual
   except:
     refs:
       - tags
+      - /^staging-[0-9]+$/
+      - /^release\/v[0-9.]+$/
```

Comment:
What about using `.feature-branch-configuration` here? I don't think we'll need to build the CI image outside of a PR.

Reply:
Perfect, I'll use it then